[docs] improve router docs before beta release (#42289)
expo/expo: docs/pages/router/advanced/modals.mdx

In the above example, the `anchor: 'index'` tells the Expo Router that it should maintain the specified anchor route in the background when presenting a modal.

## FormSheet presentation

> FormSheet presents a modal as a bottom sheet that users can drag between different heights (called detents). This is useful for content that needs partial screen coverage with interactive sizing.

Reviewer: ```suggestion
Form sheet presents a modal as a bottom sheet that app users can drag between different heights (called detents). This is useful for content that needs partial screen coverage with interactive sizing.
```

| `sheetCornerRadius`               | `number`                     | Corner radius of the sheet in pixels                     |
| `sheetLargestUndimmedDetentIndex` | `number \| 'none' \| 'last'` | Largest detent index that keeps the background undimmed  |

```tsx app/_layout.tsx
import { Stack } from 'expo-router';

export default function Layout() {
  return (
    <Stack>
      <Stack.Screen name="index" />
      <Stack.Screen
        name="modal"
        options={{
          presentation: 'formSheet',
          sheetAllowedDetents: [0.25, 0.5, 1],
          sheetInitialDetentIndex: 0,
          /* @info Shows the grabber handle at the top of the sheet (iOS only). */
          sheetGrabberVisible: true,
          /* @end */
          /* @info Sets the corner radius of the sheet. */
          sheetCornerRadius: 24,
          /* @end */
          /* @info Background stays undimmed until sheet exceeds the detent at index 1. */
          sheetLargestUndimmedDetentIndex: 1,
          /* @end */
        }}
      />
    </Stack>
  );
}
```

### Sheet footer (Android)

> > **info** `unstable_sheetFooter` is an Android-only experimental feature that may change in future releases.

Reviewer: ```suggestion
> **important** `unstable_sheetFooter` is an Android-only experimental feature that may change in future releases.
```

Let's update the type of the callout for the experimental API.

> You can add a footer to the sheet that stays visible at all detent positions:

Reviewer: ```suggestion
You can add a footer to the sheet that stays visible at all detent positions using a React component:
```

Let's add this context for clarity.

```tsx app/_layout.tsx
import { Stack } from 'expo-router';
import { View, Button } from 'react-native';

export default function Layout() {
  return (
    <Stack>
      <Stack.Screen name="index" />
      <Stack.Screen
        name="modal"
        options={{
          presentation: 'formSheet',
          sheetAllowedDetents: [0.5, 1],
          /* @info Returns a React component that renders as a footer at the bottom of the sheet. */
          unstable_sheetFooter: () => (
            <View style={{ padding: 16, backgroundColor: 'white' }}>
              <Button title="Confirm" onPress={() => {}} />
            </View>
          ),
          /* @end */
        }}
      />
    </Stack>
  );
}
```

> ### Using flex: 1 with custom detents

Reviewer: ```suggestion
### Using `flex: 1` with custom detents
```
